src/main/java/org/elasticsearch/cluster/metadata/AliasFieldsFiltering.java

```diff
@@ -0,0 +1,137 @@
+
+    public AliasFieldsFiltering(StreamInput in) throws IOException {
+        readFrom(in);
+    }
+
+    public AliasFieldsFiltering(String[] includes) {
+        this.includes = includes;
+    }
+
+    public String[] getIncludes() {
+        return includes;
+    }
+
+    public void setIncludes(String[] includes) {
+        this.includes = includes;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        AliasFieldsFiltering that = (AliasFieldsFiltering) o;
+        return Arrays.equals(includes, that.includes);
+    }
+
+    @Override
+    public int hashCode() {
+        return Arrays.hashCode(includes);
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        includes = in.readStringArray();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeStringArrayNullable(includes);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(Fields.FIELDS);
+        if (includes != null) {
+            builder.startArray(Fields.INCLUDES);
+            for (String field : includes) {
+                builder.value(field);
+            }
+            builder.endArray();
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    public static AliasFieldsFiltering fromXContext(XContentParser parser) throws IOException {
+        if (parser.currentToken() != XContentParser.Token.START_OBJECT) {
+            throw new ElasticsearchParseException("Illegal start");
+        }
+        String[] includes = Strings.EMPTY_ARRAY;
+        String fieldName = null;
+        for (XContentParser.Token token = parser.nextToken(); token != XContentParser.Token.END_OBJECT; token = parser.nextToken()) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                fieldName = parser.currentName();
+            } else if (token == XContentParser.Token.START_ARRAY) {
+                if ("includes".equals(fieldName)) {
+                    includes = readArray(parser);
+                } else {
+                    throw new ElasticsearchParseException("Unknown field [" + fieldName + "]");
+                }
+            } else {
+                throw new ElasticsearchParseException("Unknown token [" + token + "]");
+            }
+        }
+        return new AliasFieldsFiltering(includes);
+    }
+
+    private static String[] readArray(XContentParser parser) throws IOException {
+        List<String> values = new ArrayList<>();
+        while(parser.nextToken() != XContentParser.Token.END_ARRAY) {
+            if(parser.currentToken() == XContentParser.Token.VALUE_STRING) {
+                values.add(parser.text());
+            } else {
+                throw new ElasticsearchParseException("Numeric value not expected");
```

Comment:
this is a confusing message... what if it's a boolean?.... also, it's annoying to get parsing error without any constext... "expected where?"... 

Have the method accept a `String fieldName` and change the message to:

```
throw new ElasticsearchParseException("expected an array of strings for field [" + fieldName + "] but found a [" + parser.currentToken() + "] token instead");
```